src/reconstruction.jl

```diff
@@ -267,6 +270,56 @@ function exponential_decay(c::AbstractVector)
     return decay
 end
 
+function mutual_info(Xm::Vararg{<:AbstractVector,M}; k=1) where M
```

Comment:
May I suggest to move the function `mutual_info` to the `various.jl` file, which contains various mathematical functions that may (or may not) have a more general scope?